source/Calamari.Tests/Fixtures/Integration/FileSystem/CalamariPhysicalFileSystemFixture.cs

```diff
@@ -218,6 +218,127 @@ public void EnumerateFilesWithGlobShouldNotReturnTheSameFileTwice()
+        public void EnumerateFiles(string pattern, string[] expectedFilesMatchName, int expectedQty)
+        {
+            var content = "file-content" + Environment.NewLine;
+
+            var configPath = Path.Combine(rootPath, "Config");
+
+            Directory.CreateDirectory(configPath);
+            Directory.CreateDirectory(Path.Combine(configPath, "Feature1"));
+            Directory.CreateDirectory(Path.Combine(configPath, "Feature2"));
+            Directory.CreateDirectory(Path.Combine(configPath, "Feature2", "Feature2-SubFolder"));
+
+            Action<string, string, string> writeFile = (p1, p2, p3) =>
+                fileSystem.OverwriteFile(p3 == null ? Path.Combine(p1, p2) : Path.Combine(p1, p2, p3), content);
+
+            // NOTE: create all the files in *every case*, and TestCases help supply the assert expectations
+            writeFile(rootPath, "root.config", null);
+            writeFile(rootPath, "r.txt", null);
+            writeFile(configPath, "c.config", null);
+
+            writeFile(configPath, "Feature1", "f1.txt");
+            writeFile(configPath, "Feature1", "f1-a.config");
+            writeFile(configPath, "Feature1", "f1-b.config");
+            writeFile(configPath, "Feature1", "f1-c.config");
+            writeFile(configPath, "Feature2", "f2.config");
+            writeFile(configPath, "Feature1", "f2.txt");
+            writeFile(configPath, Path.Combine("Feature2", "Feature2-SubFolder"), "f2-sub.config");
+            writeFile(configPath, Path.Combine("Feature2", "Feature2-SubFolder"), "f2-sub.txt");
+
+            var resultWithoutRecursion = fileSystem.EnumerateFiles(rootPath, pattern).ToList();
+
+            resultWithoutRecursion.Should()
+                                  .HaveCount(expectedQty, $"{pattern} should have found {expectedQty}, but found {resultWithoutRecursion.Count}");
+            foreach (var expectedFileName in expectedFilesMatchName)
+            {
+                resultWithoutRecursion.Should()
+                                      .Contain(r => Path.GetFileName(r) == expectedFileName, $"{pattern} should have found {expectedFileName}, but didn't");
+            }
+        }
+
+        [Test]
+        [TestCase(@"*.txt", new [] {"r.txt", "f1.txt", "f2.txt", "f2-sub.txt"}, 4)]
+        [TestCase(@"*.config", new [] { "root.config", "c.config", "f1-a.config", "f1-b.config", "f1-c.config", "f2.config", "f2-sub.config"}, 7)]
+        [TestCase(@"Config/*.config",  new [] { "c.config", "f1-a.config", "f1-b.config", "f1-c.config", "f2.config", "f2-sub.config"}, 6)]
+        [TestCase(@"Config/Feature1/*.config",new[] {"f1-a.config", "f1-b.config", "f1-c.config"}, 3)]
+        [TestCase(@"Config/Feature2/*.config", new[] {"f2.config", "f2-sub.config"}, 2)]
+        [TestCase(@"Config/Feature2/Feature2-SubFolder/*.config",new[] {"f2-sub.config"}, 1)]
+        public void EnumerateFilesRecursively(string pattern, string[] expectedFilesMatchNameWithRecursion, int expectedQty)
```

Comment:
I could combine this and the above non-recursive search and pass the file enumeration method as a parameter, but I decided to stick with this primarily for separation and clarity. Let me know if you'd rather them combined.